google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -175,25 +179,48 @@ private static void generateAppYaml(@NotNull AppEngineFlexibleFacet facet,
 
   private static void setupDeploymentRunConfiguration(Module module) {
     RunManager runManager = RunManager.getInstance(module.getProject());
-    AppEngineCloudType serverType =
-        ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
-    DeployToServerConfigurationType configurationType
-        = DeployToServerConfigurationTypesRegistrar.getDeployConfigurationType(serverType);
-
-    RunnerAndConfigurationSettings settings = runManager.createRunConfiguration(
-        configurationType.getDisplayName(), configurationType.getFactory());
-
-    // Sets the GAE Flex server, if any exists, in the run config.
-    DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration> runConfiguration =
-        (DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration>)
-            settings.getConfiguration();
-    RemoteServer<AppEngineServerConfiguration> server =
-        ContainerUtil.getFirstItem(RemoteServersManager.getInstance().getServers(serverType));
-    if (server != null) {
-      runConfiguration.setServerName(server.getName());
+
+    if (!hasFlexibleDeploymentConfiguration(runManager.getAllConfigurationsList())) {
+      AppEngineCloudType serverType =
+          ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
+      DeployToServerConfigurationType configurationType
+          = DeployToServerConfigurationTypesRegistrar.getDeployConfigurationType(serverType);
+
+      RunnerAndConfigurationSettings settings = runManager.createRunConfiguration(
+          configurationType.getDisplayName(), configurationType.getFactory());
+
+      // Sets the GAE Flex server, if any exists, in the run config.
+      DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration> runConfiguration =
+          (DeployToServerRunConfiguration<?, AppEngineDeploymentConfiguration>)
+              settings.getConfiguration();
+      RemoteServer<AppEngineServerConfiguration> server =
+          ContainerUtil.getFirstItem(RemoteServersManager.getInstance().getServers(serverType));
+      if (server != null) {
+        runConfiguration.setServerName(server.getName());
+      }
+
+      runManager.addConfiguration(settings, false /* shared */);
     }
+  }
+
+  private static boolean hasFlexibleDeploymentConfiguration(List<RunConfiguration> runConfigs) {
+    return runConfigs
+        .stream()
+        .anyMatch(runConfig -> {
```

Comment:
I kind of like the flatMap(select()) style in the last answer here: http://stackoverflow.com/questions/35740543/java-8-stream-check-if-instanceof

It's a shame there isn't a utility method for this in the JDK.